main.cc

```diff
@@ -114,7 +114,7 @@ class Except : public std::exception {
 		str_ = msg;
 
 		// Add system error code (errno).
-		if (errno) {
+		if (errno && errno != EINPROGRESS && errno != EAGAIN) {
```

Comment:
Let `Except` handle even valid errno codes: at this point we already thrown an exception and the program is going to terminate - in this case we at least see that it was a wrong exception and will fix it.